File: support/mock_backend/wiringPi/w25q64.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "wiringPi/w25q64.h"
void W25Q64_begin(uint8_t cs) { (void)cs; }
uint8_t W25Q64_readStatusReg1(void) { return 1; }
uint8_t W25Q64_readStatusReg2(void) { return 1; }
void W25Q64_readManufacturer(uint8_t * d) { d[0]=0xEF;d[1]=0x40;d[2]=0x17; }
void W25Q64_readUniqieID(uint8_t * d) { for(int i=0;i<7;++i)d[i]=(uint8_t)(0x10+i); }
bool W25Q64_IsBusy(void) { return true; }
void W25Q64_powerDown(void) {  }
void W25Q64_WriteEnable(void) {  }
void W25Q64_WriteDisable(void) {  }
uint16_t W25Q64_read(uint32_t addr, uint8_t * buf, uint16_t n) { for(uint16_t i=0;i<n;++i)buf[i]=(uint8_t)(addr+i);return n; }
uint16_t W25Q64_fastread(uint32_t addr, uint8_t * buf, uint16_t n) { for(uint16_t i=0;i<n;++i)buf[i]=(uint8_t)(addr+i);return n; }
bool W25Q64_eraseSector(uint16_t sect_no, bool flgwait) { (void)sect_no;(void)flgwait;return true; }
bool W25Q64_erase64Block(uint16_t blk_no, bool flgwait) { (void)blk_no;(void)flgwait;return true; }
bool W25Q64_erase32Block(uint16_t blk_no, bool flgwait) { (void)blk_no;(void)flgwait;return true; }
bool W25Q64_eraseAll(bool flgwait) { (void)flgwait;return true; }
uint16_t W25Q64_pageWrite(uint16_t sect_no, uint16_t inaddr, uint8_t * data, uint8_t n) { (void)sect_no;(void)inaddr;(void)data;return n; }

File: tests/wiringPi/w25q64.py
import importlib
import pytest

m=importlib.import_module("wiringop.wiring_pi.w25q64")

def test_w25q64_begin():
    _=m.w25q64_begin(1)

def test_w25q64_read_status_reg1():
    _=m.w25q64_read_status_reg1()

def test_w25q64_read_status_reg2():
    _=m.w25q64_read_status_reg2()

def test_w25q64_read_manufacturer():
    assert m.w25q64_read_manufacturer()==bytes.fromhex("ef4017")

def test_w25q64_read_unique_id():
    assert m.w25q64_read_unique_id()==bytes(range(0x10,0x17))

def test_w25q64_is_busy():
    _=m.w25q64_is_busy()

def test_w25q64_power_down():
    _=m.w25q64_power_down()

def test_w25q64_write_enable():
    _=m.w25q64_write_enable()

def test_w25q64_write_disable():
    _=m.w25q64_write_disable()

def test_w25q64_read():
    assert m.w25q64_read(0x20,4)==bytes([0x20,0x21,0x22,0x23])

def test_w25q64_fast_read():
    assert m.w25q64_fast_read(0x20,4)==bytes([0x20,0x21,0x22,0x23])

def test_w25q64_erase_sector():
    _=m.w25q64_erase_sector(1, True)

def test_w25q64_erase64_block():
    _=m.w25q64_erase64_block(1, True)

def test_w25q64_erase32_block():
    _=m.w25q64_erase32_block(1, True)

def test_w25q64_erase_all():
    _=m.w25q64_erase_all(True)

def test_w25q64_page_write():
    assert m.w25q64_page_write(0,0,b"abc")==3
    with pytest.raises(ValueError):m.w25q64_page_write(0,250,b"0123456789")
